sdk/identity/Azure.Identity/src/ClientAssertionCredential.cs

```diff
@@ -12,26 +12,57 @@
 
 namespace Azure.Identity
 {
-    internal class ClientAssertionCredential : TokenCredential
+    /// <summary>
+    /// Enables authentication of an AAD service principal using a signed client assertion.
+    /// </summary>
+    public class ClientAssertionCredential : TokenCredential
     {
         internal string TenantId { get; }
         internal string ClientId { get; }
         internal MsalConfidentialClient Client { get; }
         internal bool AllowMultiTenantAuthentication { get; }
 
-        public ClientAssertionCredential(string tenantId, string clientId, Func<Task<string>> getAssertionCallback, ClientAssertionCredentialOptions options = default) :
-            this(tenantId, clientId, () => getAssertionCallback().GetAwaiter().GetResult(), options)
+        /// <summary>
+        /// Protected constructor for mocking.
+        /// </summary>
+        protected ClientAssertionCredential()
+        { }
+
+        /// <summary>
+        /// Creates an instance of the ClientCertificateCredential with the details needed to authenticate against Azure Active Directory using a signed client assertion.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="assertionCallback">An asynchronous callback returning a valid client assertion used to authenticate the service principal.</param>
+        /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
+        public ClientAssertionCredential(string tenantId, string clientId, Func<CancellationToken, Task<string>> assertionCallback, ClientAssertionCredentialOptions options = default)
         {
+            TenantId = tenantId;
+            ClientId = clientId;
+            Client = options?.MsalClient ?? new MsalConfidentialClient(options?.Pipeline ?? CredentialPipeline.GetInstance(options), tenantId, clientId, assertionCallback, null, null, options?.IsLoggingPIIEnabled ?? false);
         }
 
-        public ClientAssertionCredential(string tenantId, string clientId, Func<string> getAssertionCallback, ClientAssertionCredentialOptions options = default)
+        /// <summary>
+        /// Creates an instance of the ClientCertificateCredential with the details needed to authenticate against Azure Active Directory using a signed client assertion.
```

Comment:
Should we call out more in the summary that the difference between these ctors is the sync/async callback?